localstack-core/localstack/services/apigateway/next_gen/execute_api/gateway_response.py

```diff
@@ -1,125 +1,143 @@
-from localstack.aws.api.apigateway import GatewayResponseType
+from functools import cache
 
+from localstack.aws.api.apigateway import (
+    GatewayResponse,
+    GatewayResponseCode,
+    GatewayResponseType,
+)
 
-class BaseGatewayResponse(Exception):
+DEFAULT_GATEWAY_RESPONSE = {
+    "responseParameters": {},
+    "responseTemplates": {"application/json": '{"message":$context.error.messageString}'},
+}
+
+
+@cache
+def build_default_response(type: GatewayResponseType) -> dict:
+    response = GatewayResponse(**DEFAULT_GATEWAY_RESPONSE, responseType=type, defaultResponse=True)
+    if status_code := GatewayResponseCode[type]:
+        response["statusCode"] = status_code
+    return response
```

Comment:
I was trying to avoid making copies of the dict everywhere, when I assume most users barely never modify it or will only modify some elements of it.

But yeah, we could make the code simpler by creating a `deepcopy` of it when creating a new api. Then each deployment would get a snapshot of the current full dict representation. It does make my new utility a little bit superfluous then. 

I just didn't like very much the big hardcoded dict in there! :smile: